### PR TITLE
Deprecate unused compat functions

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Deprecates:
- Deprecate `conda.activate.ensure_binary`
- Deprecate `conda.activate.ensure_fs_path_encoding`
- Deprecate `conda.common.compat.six_with_metaclass`
- Deprecate `conda.common.compat.open`
- Deprecate `conda.common.compat.ensure_unicode`
- Deprecate `conda.common.compat.ensure_fs_path_encoding`
- Deprecate `conda.common.compat.FILESYSTEM_ENCODING`

### Checklist - did you ...

- [x] Add a file to the `news` directory ([using the template](https://github.com/conda/conda/blob/main/news/TEMPLATE)) for the next release's release notes?
     
- [x] Add / update necessary tests?
- [ ] Add / update outdated documentation?

### CODE REVIEW

File: conda/base/constants.py [hunk 1]
```diff
@@ -247,7 +247,7 @@ def __str__(self):
         return f"{self.value}"
 
 
-class ChannelPriority(six_with_metaclass(ChannelPriorityMeta, ValueEnum)):
+class ChannelPriorityNew(ValueEnum, metaclass=ChannelPriorityMeta):
```
Comment (kenodegard) [inline conda/base/constants.py:250]:
```suggestion
class ChannelPriority(ValueEnum, metaclass=ChannelPriorityMeta):
```

File: news/14144-deprecate-unused-compat [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+### Bug fixes
+
+* <news item>
+
+### Deprecations
+
+Mark `conda.activate.ensure_binary` as pending deprecation. Use `conda.common.compat.ensure_binary` instead. (#14144)
+Mark `conda.activate.ensure_fs_path_encoding` as pending deprecation. (#14144)
+Mark `conda.common.compat.six_with_metaclass` as pending deprecation. Use class' `metadata=` keyword argument instead. (#14144)
```
Comment (beeankha) [inline news/14144-deprecate-unused-compat:13]:
```suggestion
Mark `conda.common.compat.six_with_metaclass` as pending deprecation. Use class' `metaclass=` keyword argument instead. (#14144)
```

**Review by beeankha — APPROVED** [review]
Other than a small change in the news file, this LGTM!

**Review by jezdez — DISMISSED** [review]

File: conda/common/compat.py [hunk 3]
```diff
@@ -80,6 +86,7 @@ def open(
         )
 
 
+@deprecated("25.3", "25.9", addendum="Use class' `metadata=` keyword argument instead.")
```
Comment (kenodegard) [inline conda/common/compat.py:89]:
```suggestion
@deprecated("25.3", "25.9", addendum="Use class' `metaclass=` keyword argument instead.")
```

File: news/14144-deprecate-unused-compat (hunk 2 above)
Comment (kenodegard) [reply news/14144-deprecate-unused-compat]: ah good catch!

File: conda/common/compat.py [hunk 4]
```diff
@@ -80,6 +86,7 @@ def open(
         )
 
 
+@deprecated("25.3", "25.9", addendum="Use class' `metaclass=` keyword argument instead.")
```
Comment (kenodegard) [inline conda/common/compat.py:89]:
```suggestion
@deprecated(
    "25.3",
    "25.9",
    addendum="Use class' `metaclass=` keyword argument instead.",
)
```

**Review by beeankha — APPROVED** [review]

**Pull request merged**